provisioning/device/src/interfaces.ts

```diff
@@ -90,20 +109,21 @@ export interface RegistrationClient {
   /**
    * Register the device with the provisioning service
    */
-  register(registrationId: string, forceRegistration: boolean, callback: (err?: Error, result?: any) => void): void;
+  register(request: RegistrationRequest, callback: (err?: Error, result?: any) => void): void;
   /**
    * Cancel the registration process if it is in progress.
    */
   cancel(callback: (err?: Error) => void): void;
 }
 
 /**
- * Device configuration returned when registration using TPM is complete
+ * Information passed between client and transport during Tpm registration
  */
 export interface TpmRegistrationInfo {
```

Comment:
>TpmRegistrationInfo [](start = 17, length = 19)

registrationId is going to be duplicated in this structure - let's remove it from TpmRegistrationInfo alltogether